### ISSUE TITLE
Good grief, I am writing an issue to myself

### ISSUE THREAD

**Issue by escaip** [escaip/simple-nn#1]
I'm not sure what kind of magic is happening in my neural network, but it looks like the network just randomly chooses whether to behave well or not. The gradient checking algorithm shows a great difference between numerically calculated gradients and backpropagated ones. Yet the algorithm works (at least sometimes).  I'll leave it for a couple of days...
